src/test/java/com/vozniuk/bookstore/controller/BookControllerTest.java

```diff
@@ -0,0 +1,218 @@
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.vozniuk.bookstore.dto.book.BookDto;
+import com.vozniuk.bookstore.dto.book.CreateBookRequestDto;
+import com.vozniuk.bookstore.model.Book;
+import com.vozniuk.bookstore.model.Category;
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+import org.testcontainers.shaded.org.apache.commons.lang3.builder.EqualsBuilder;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+class BookControllerTest {
+    protected static MockMvc mockMvc;
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @BeforeAll
+    static void beforeAll(
+            @Autowired WebApplicationContext applicationContext
+    ) {
+        mockMvc = MockMvcBuilders
+                .webAppContextSetup(applicationContext)
+                .apply(springSecurity())
+                .build();
+    }
+
+    @Test
+    @WithMockUser(username = "user", roles = {"USER"})
+    @DisplayName("Verify getAll() method works correctly")
+    @Sql(scripts = "classpath:database/books/add-default-books-to-books-table.sql",
+            executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+    @Sql(scripts = "classpath:database/books/remove-all-books-from-books-table.sql",
+            executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD)
+    void getAll_NonEmptyRepository_ReturnsAllBooks() throws Exception {
+        List<BookDto> expected = new ArrayList<>();
+        expected.add(new BookDto().setId(1L).setTitle("Java Book 1").setAuthor("Some author 1")
+                .setIsbn("9781617290459").setPrice(BigDecimal.valueOf(499))
+                .setCategoryIds(new HashSet<Long>(0)));
+        expected.add(new BookDto().setId(2L).setTitle("Java Book 2").setAuthor("Some author 2")
```

Comment:
Code duplication. I suggest to create builder for books dto in more clear and beautiful way, also It will be great if repeatable books will be constants